### ISSUE TITLE
LabelSmoothingLoss (loss_fn="LS") doesn't apply label smoothing — identical to plain BCE

### ISSUE THREAD

**Issue by Demirrr** [dice-group/dice-embeddings#453]
## Problem

\`LabelSmoothingLoss.forward\` in \`dicee/losses/custom_losses.py\` is:

\`\`\`python
def forward(self, pred, target):
    criterion = torch.nn.BCEWithLogitsLoss()
    final_loss = criterion(pred, target)
    return final_loss
\`\`\`

This never uses \`self.smoothness_ratio\` (set in \`__init__\` from the \`label_smoothing_rate\` config option) at all — it's byte-for-byte identical to \`DefaultBCELoss.forward\`. Selecting \`loss_fn="LS"\` silently trains with plain BCE, no smoothing applied, regardless of \`label_smoothing_rate\`.

Same root issue likely affects \`CombinedLSandLR\` and \`AggregatedLSandLR\`, which both delegate to \`LabelSmoothingLoss\` for their "smoothing" branch/term.

Found while documenting these classes for #444.

## Context
Note \`label_smoothing_rate\` is *also*, separately, applied correctly at the dataset level (\`dicee/dataset_classes/_label_based.py\`, \`_negative_sampling.py\` bake smoothing directly into the training targets, independent of \`loss_fn\`) — that path is unaffected. This issue is specifically about the \`loss_fn="LS"\` mechanism being a no-op.

## Suggested fix
Implement smoothing in \`LabelSmoothingLoss.forward\`, e.g. following the same KL-divergence-based pattern already used correctly in \`AdaptiveLabelSmoothingLoss.forward\`:
\`\`\`python
pred = F.log_softmax(logits, dim=-1)
smoothed_target = (1 - self.smoothness_ratio) * target + self.smoothness_ratio * (1 - target) / (num_classes - 1)
return F.kl_div(pred, smoothed_target, reduction="batchmean")
\`\`\`
(or whatever formula the original author intended — worth confirming with them if possible). Add a test asserting the loss changes as \`smoothness_ratio\` varies, to prevent this regressing silently again.

**Comment by adelmemariani** [comment]
https://github.com/dice-group/dice-embeddings/blob/0065bcc9f2fb1ce2d656963353ab53ae787a1860/dicee/losses/custom_losses.py#L39


https://github.com/dice-group/dice-embeddings/blob/0065bcc9f2fb1ce2d656963353ab53ae787a1860/dicee/losses/custom_losses.py#L42


the above links are about the original implementation, maybe they can help.

**Comment by adelmemariani** [comment]
The loss in my branch is:

`class LabelSmoothingLoss(nn.Module):`

```
    def __init__(self, smoothness_ratio=0.0):
        super(LabelSmoothingLoss, self).__init__()
        self.smoothness_ratio = smoothness_ratio
```

```
    def forward(self, pred, target):
            criterion = torch.nn.BCEWithLogitsLoss()
            final_loss = criterion(pred, target)
  
            return final_loss
```

**Comment by adelmemariani** [comment]
Then in base_model.py:

```
        if self.args["loss_fn"] == "LS":
            self.loss = LabelSmoothingLoss(smoothness_ratio=self.args["label_smoothing_rate"])
```

**Comment by adelmemariani** [comment]
Then I run it with:


```
if EXP_TYPE == "LS":
                    args.loss_fn = "BCELoss"
                    args.label_smoothing_rate = 0.1
                    args.adaptive_swa = False
```
